volplugin/volplugin.go

```diff
@@ -95,3 +110,45 @@ func logHandler(name string, debug bool, actionFunc func(http.ResponseWriter, *h
 		actionFunc(w, r)
 	}
 }
+
+func (dc *DaemonConfig) updateMounts() error {
+	cd := ceph.NewDriver()
+	mounts, err := cd.Mounted()
+	if err != nil {
+		return err
+	}
+
+	for _, mount := range mounts {
+		parts := strings.Split(mount.Volume.Name, "/")
+		if len(parts) != 2 {
+			log.Warnf("Invalid volume named %q in ceph scan: skipping refresh", mount.Volume.Name)
+			continue
+		}
+
+		log.Infof("Refreshing existing mount for %q", mount.Volume.Name)
+
+		volConfig, err := requestVolumeConfig(dc.Master, parts[0], parts[1])
+		if err != nil {
```

Comment:
> I'm not entirely sure if we should be handling those at all or taking action to reconcile the system with what we perceive is reality by removing the mounts.

I would not try to handle this if it's not going to be a common case. If we can assume that ceph is managed by volplugin it will simplify error handling.